tests/test_hpack.c

```diff
@@ -876,3 +888,357 @@ static int test_hpack_dynamic_table_size_update_from_setting(struct aws_allocato
+                                                headers to check the result */
+        for (size_t j = 0; j < num_headers_to_make; j++) {
+            char test_header_str[256];
+            /**
+             * - Don't use _ or - between word, the response will capitalize the first char of every word :(
+             * - Don't use _, the response will change it to -. :(
+             */
+            uint64_t random_64_bit_num = 0;
+            aws_device_random_u64(&random_64_bit_num);
+
+            size_t headers = (size_t)random_64_bit_num % headers_pool_size;
+            sprintf(test_header_str, "crttest-%zu", headers);
+            char test_value_str[256];
+            size_t value = (size_t)random_64_bit_num % values_pool_size;
+            sprintf(test_value_str, "value-%zu", value);
+            struct aws_byte_cursor existed_value;
+            if (aws_http_headers_get(test_headers, aws_byte_cursor_from_c_str(test_header_str), &existed_value) ==
+                AWS_OP_SUCCESS) {
+                /* If the header has the same name already exists in the headers, the response will combine the values
+                 * together. Do the same thing for the header to check. */
+                char combined_value_str[1024];
+                sprintf(combined_value_str, "" PRInSTR ",%s", AWS_BYTE_CURSOR_PRI(existed_value), test_value_str);
+                aws_http_headers_set(
+                    test_headers,
+                    aws_byte_cursor_from_c_str(test_header_str),
+                    aws_byte_cursor_from_c_str(combined_value_str));
+            } else {
+                aws_http_headers_add(
+                    test_headers,
+                    aws_byte_cursor_from_c_str(test_header_str),
+                    aws_byte_cursor_from_c_str(test_value_str));
+            }
+            aws_http_headers_add(
+                request_headers,
+                aws_byte_cursor_from_c_str(test_header_str),
+                aws_byte_cursor_from_c_str(test_value_str));
+        }
+        struct aws_byte_buf echo_body;
+        ASSERT_SUCCESS(aws_byte_buf_init(&echo_body, allocator, 100));
+        struct aws_http_make_request_options request_options = {
+            .self_size = sizeof(request_options),
+            .request = request,
+            .user_data = &echo_body,
+            .on_response_body = s_tester_on_echo_body,
+            .on_complete = s_tester_on_stream_completed,
+        };
+        struct aws_http_stream *stream = aws_http_connection_make_request(s_tester.connection, &request_options);
+        ASSERT_NOT_NULL(stream);
+        aws_http_stream_activate(stream);
+        aws_http_stream_release(stream);
+
+        /* Wait for the stream to complete */
+        ASSERT_SUCCESS(s_wait_on_streams_completed_count(1));
+        --s_tester.stream_completed_count;
+        if (!s_tester.stream_completed_with_200) {
```

Comment:
have 4xx error codes fail the test, it means the server is telling us that our request was bad